src/handlers.rs

```diff
@@ -54,22 +62,25 @@ impl<S> FromRequest<S> for HawkHeader {
     }
 }
 
-macro_rules! endpoint {
+macro_rules! db_endpoint {
     ($handler:ident: $data:ident ($path:ident: $path_type:ty $(, $param:ident: $type:ty)*) {$($property:ident: $value:expr,)*}) => {
         pub fn $handler(
-            ($path, _state$(, $param)*): (Path<$path_type>, State<ServerState>$(, $type)*),
+            ($path, state$(, $param)*): (Path<$path_type>, State<ServerState>$(, $type)*),
         ) -> FutureResponse<HttpResponse> {
-            let _data = data::$data {
-                $($property: $value,)*
-            };
-            Box::new(result(Ok(HttpResponse::Ok().json(()))))
+            Box::new(
+                state.db.$handler(params::$data {
+                    $($property: $value,)*
+                })
+                .map_err(From::from)
+                .map(|result| HttpResponse::Ok().json(result))
```

Comment:
Fwiw this macro implementation is not going to work long-term because some of the endpoints need to do extra marshalling, e.g. setting `payload_size` on BSOs and so on.

Reply:
(or we could keep this simple and do that stuff inside the db layer, but it would be common to both implementations so outside seems more appropriate)

Reply:
Oh yeah, that is a splendid suggestion, thanks!